appservice/src/intTest/groovy/nu/yona/server/DeviceTest.groovy

```diff
@@ -178,16 +182,94 @@ class DeviceTest extends AbstractAppServiceIntegrationTest
 					assertResponseStatus(it, 400)
 					assert it.responseData.code == "error.device.invalid.device.name"
 				}, "John", "Doe", "JD",
-				makeMobileNumber(ts), "some:thing", "IOS")
+				makeMobileNumber(ts), "some:thing", "IOS", Device.SUPPORTED_APP_VERSION)
 
 		then:
 		assert johnAsCreated == null // Creation failed
 	}
 
+	def 'Richard posts empty app opened event'()
+	{
+		given:
+		def richard = addRichard()
+
+		when:
+		def response = appService.createResourceWithPassword(richard.postOpenAppEventUrl, "{}", richard.password)
+
+		then:
+		assertResponseStatusOk(response)
+
+		cleanup:
+		appService.deleteUser(richard)
+	}
+
+	def 'Richard posts app opened event with a valid app version'()
```

Comment:
"valid app version"-> "valid operating system and app version"

Reply:
Done.